doc/source/development/maintaining.rst

```diff
@@ -345,7 +345,126 @@ The benchmarks are scheduled by Airflow. It has a dashboard for viewing and debu
 Release process
 ---------------
 
-The process for releasing a new version of pandas can be found at https://github.com/pandas-dev/pandas-release
+The release process makes a snapshot of pandas (a git commit) available to users with
+a particular version number. After the release the new pandas version will be available
+in the next places:
+
+- Git repo with a new tag
+- Source distribution in a [GitHub release](https://github.com/pandas-dev/pandas/releases)
+- Pip packages in the [PyPI](https://pypi.org/project/pandas/)
+- Conda/Mamba packages in [conda-forge](https://anaconda.org/conda-forge/pandas)
+
+The process for releasing a new version of pandas is detailed next.
+
+The instructions contain ``<version>`` which needs to be replaced with the version
+to be released (e.g. ``1.5.2``). Also the branch to be released ``<branch>``, which
+depends on whether the version being released is the release candidate of a new version,
+or any other version. Release candidates are released from ``main``, while other
+versions are released from their branch (e.g. ``1.5.x``).
+
+
+Pre-release
```

Comment:
It would be good to have a section on cutting new branches, detailing what needs to be updated in the CI and when to do it(I don't remember if we cut during release, or do it before the release and start backporting).